### PR TITLE
AtlasRDD to Dataframe conversion methods

### PR DESCRIPTION
**Pull request by jwpgage**
### Description:

A suite of helper methods that convert an atlasRDD into Dataframes, with one Dataframe for each entity type.

### Potential Impact:

Allows for querying atlases within a spark session.

### Unit Test Approach:

~~Test to be added.~~

Test deserialization and data frame creation.

### Test Results:

Dataframes successfully generated.

------

In doubt: [Contributing Guidelines](https://github.com/osmlab/atlas/blob/dev/CONTRIBUTING.md)

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/generator/AtlasDataFrame.java [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+        final int line = 3;
+        final int point = 4;
+        final int relation = 5;
+        switch (value)
+        {
+            case 0:
+                return "NODE";
+            case 1:
+                return "EDGE";
+            case 2:
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/generator/AtlasDataFrame.java:362]: Aah checkstyle! Mabe make it consistent with the ones below?

File: src/test/java/org/openstreetmap/atlas/generator/AtlasDataFrameTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ */
+public class AtlasDataFrameTest
+{
+    private static final Logger logger = LoggerFactory.getLogger(AtlasDataFrameTest.class);
+
+    @Test
+    public void atlasDataFrameTest()
+    {
+        final Resource file = new InputStreamResource(
+                () -> new File("/Users/jamesgage/Desktop/scratch/dfTesting/DMA_9-168-233.atlas")
```
Comment (matthieun) [inline src/test/java/org/openstreetmap/atlas/generator/AtlasDataFrameTest.java:32]: This should become a file in the resources folder. To check it in, make sure it is as small as necessary!

File: src/main/java/org/openstreetmap/atlas/generator/AtlasDataFrame.java [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+import scala.collection.JavaConversions;
+
+/**
+ * Methods for converting an {@link Atlas} RDD to {@link Dataset}
+ * 
+ * @author jamesgage
+ */
+public final class AtlasDataFrame
+{
+    protected static Dataset<Row> atlasAreasToDataFrame(final JavaRDD<Atlas> atlasRDD,
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/generator/AtlasDataFrame.java:28]: How about making all those methods public?

**Review by matthieun — CHANGES_REQUESTED** [review]
Nice! Just a few nits here and there!
Also it would be nice to have a readme with it that details what this does, as well as defines the schema that each row has!

File: src/main/java/org/openstreetmap/atlas/generator/AtlasDataFrame.java (hunk 1 above)
Comment (jwpgage) [reply src/main/java/org/openstreetmap/atlas/generator/AtlasDataFrame.java]: yes, definitely a checkstyle-ism :)

**Review by lucaspcram — DISMISSED** [review]

File: src/test/java/org/openstreetmap/atlas/generator/dataframe/AtlasDataFrameTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+                javaSparkContext);
+        final Dataset<Row> relationDataframe = AtlasDataFrame
+                .atlasRelationsToDataFrame(atlasJavaRDD, javaSparkContext);
+
+        Assert.assertEquals("24433389000000", areaDataframe.first().get(0));
+        Assert.assertEquals("27989500000000", edgeDataframe.first().get(0));
+        Assert.assertEquals("307351652000000", nodeDataframe.first().get(0));
+        Assert.assertEquals("99202295000000", lineDataframe.first().get(0));
+        Assert.assertEquals("4553243887000000", pointDataframe.first().get(0));
+        Assert.assertEquals("9451753000000", relationDataframe.first().get(0));
```
Comment (matthieun) [inline src/test/java/org/openstreetmap/atlas/generator/dataframe/AtlasDataFrameTest.java:66]: Is the order of the results in the dataset always deterministic?
Maybe also do a check on the size of each Dataset?

File: src/main/java/org/openstreetmap/atlas/generator/dataframe/README.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# `AtlasDataFrame`
+## Overview
+`AtlasDataFrame` is a class that houses methods that convert an atlas RDD to a `DataFrame`. This allows for easily querying the data of an atlas from within an active spark session. Each entity is separated into its own dataframe, and supports its own schema.
+
+### Schemas
+
+Areas:
+
+`identifier` `geometry` `bounds` `tags` `relations`
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/generator/dataframe/README.md:9]: Here I would make that a markdown table, with the type below, like:
`identifier` | `geometry` | `bounds` | `tags` | `relations`
--|--|--|--|--
String|String|String|Map|Array

And the same for the ones below.

File: src/main/java/org/openstreetmap/atlas/generator/dataframe/README.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# `AtlasDataFrame`
```
Comment (matthieun) [inline src/main/java/org/openstreetmap/atlas/generator/dataframe/README.md:1]: Can you add a relative link to this page here? https://github.com/osmlab/atlas/blob/dev/README.md#whats-in-it

File: src/test/java/org/openstreetmap/atlas/generator/dataframe/AtlasDataFrameTest.java (hunk 4 above)
Comment (matthieun) [reply src/test/java/org/openstreetmap/atlas/generator/dataframe/AtlasDataFrameTest.java:66]: Might also be nice to test the inside of the whole row... on top of the ID.

File: src/main/java/org/openstreetmap/atlas/generator/dataframe/README.md (hunk 6 above)
Comment (jwpgage) [reply src/main/java/org/openstreetmap/atlas/generator/dataframe/README.md:1]: Does the relative link need to be from within the same repo? It seems that all the others are contained within atlas

File: src/test/java/org/openstreetmap/atlas/generator/dataframe/AtlasDataFrameTest.java (hunk 4 above)
Comment (jwpgage) [reply src/test/java/org/openstreetmap/atlas/generator/dataframe/AtlasDataFrameTest.java:67]: Added a check for size and the tags, but haven't tested the long strings for geometry etc

File: src/main/java/org/openstreetmap/atlas/generator/dataframe/README.md (hunk 6 above)
Comment (matthieun) [reply src/main/java/org/openstreetmap/atlas/generator/dataframe/README.md:1]: Ah right! Different repo haha! Maybe we need to have it linked in the atlas-generator's main readme?

Comment (matthieun) [reply src/main/java/org/openstreetmap/atlas/generator/dataframe/README.md:1]: That's probably for another PR though. We need to add more discoverability there..

**Review by matthieun — APPROVED** [review]

**Review by lucaspcram — APPROVED** [review]

**Pull request merged**